packages/compass-collection/src/components/mock-data-generator-modal/document-count-screen.tsx

```diff
@@ -67,20 +67,40 @@ const DocumentCountScreen = ({
   onDocumentCountChange,
   schemaAnalysisState,
 }: Props) => {
+  const documentCountNumber = parseInt(documentCount, 10);
+
   const estimatedDiskSize = useMemo(
     () =>
       schemaAnalysisState.status === 'complete' &&
-      schemaAnalysisState.schemaMetadata.avgDocumentSize
+      schemaAnalysisState.schemaMetadata.avgDocumentSize &&
+      !isNaN(documentCountNumber)
         ? formatBytes(
-            schemaAnalysisState.schemaMetadata.avgDocumentSize * documentCount
+            schemaAnalysisState.schemaMetadata.avgDocumentSize *
+              documentCountNumber
           )
         : 'Not available',
-    [schemaAnalysisState, documentCount]
+    [schemaAnalysisState, documentCountNumber]
   );
 
-  const isOutOfRange = documentCount < 1 || documentCount > MAX_DOCUMENT_COUNT;
+  const isInputEmpty = documentCount.trim() === '';
```

Comment:
If we want to differentiate between an empty input and an invalid input, let's make sure this doesn't unintentionally count non-empty `NaN` values as empty. I think that right it shows the empty input error message when you type non-number string.

`const isInputEmpty = !isNaN(documentCountNumber) && documentCount.trim() === '';`

Reply:
I think this went away after you updated `isInputInvalid`.